packages/foundation-models/src/azure-openai/azure-openai-chat-client.test.ts

```diff
@@ -75,4 +75,93 @@ describe('Azure OpenAI chat client', () => {
 
     await expect(client.run(prompt)).rejects.toThrow('status code 400');
   });
+
+  it('executes a request with the custom request group', async () => {
```

Comment:
It is important that the RG is used for both: 

* lookup of deployment ID
* inference request

Since we are already testing the latter here, I'd consider also covering the former. While this does come with a bit more request mocking, I feel that we do need a test for this.

We could also invert the strategy, always covering both requests and only in one single test case implement (pseudo code):
```ts
new AzureOpenAiChatClient({ deploymentId: '1234' })
expect(nock).toNeverBeCalled()
```

Arguably we should also cover that
```
new AzureOpenAiChatClient({ deploymentId: '1234', resourceGroup: 'foo' });
```
works, i.e. that the RG `foo` is sent for the inference request, but no deployment ID lookup happens..

Reply:
I generally stronlgy agree, however, mocking became extremely annoying with ESM in Jest.
I am currently looking into a jest plugin called [`testdouble`](https://github.com/testdouble/testdouble-jest) which might enable us to use mocking the way you describe it again.
In commonJS we would be able to simply use jest.spyOn(), this however does only work on instantiated classes in ESM out of the box.

Reply:
After trying every possible combination, I came to the conclusion that we can't use it which means mocking is not realistically feasible for this scenario 🙄

Reply:
I think we have a long list of test for the former in `deployment-resolver.test.ts`

Reply:
> mocking is not realistically feasible for this scenario

🫠

> long list of tests for the former

They don't really test:

> that the RG is used for both:
> * lookup of deployment ID
> * inference request

I.e. there is no assertion that guarantees "if the deployment ID is resolved using RG `foo`, then the inference request also contains RG `foo`"

Reply:
I see, I've added a test for this now to the azure client as well as orchestration 👍